compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/ScenarioServiceMain.scala

```diff
@@ -18,8 +20,8 @@ object ScenarioServiceMain extends App {
   // default to 128MB
   val maxMessageSize = args.headOption.map(_.toInt).getOrElse(128 * 1024 * 1024)
   val server =
-    ServerBuilder
-      .forPort(0) // any free port
+    NettyServerBuilder
+      .forAddress(new InetSocketAddress("127.0.0.1", 0)) // any free port
```

Comment:
Can I propose `InetAddress.getLoopbackAddress.getHostName`? It'll return `"localhost"`.

Reply:
Oh, actually, you can pass an `InetAddress` to `InetSocketAddress`, so you can do:

```suggestion
      .forAddress(new InetSocketAddress(InetAddress.getLoopbackAddress, 0)) // any free port
```

(Don't apply this; it won't work because imports.)

Reply:
iirc we’ve had some issues with using `localhost` on the scenario service in our alpine docker container (glibc, musl dns crap). I _think_ that was on the client side but it would be good to confirm that before.

Reply:
Is it documented to always return localhost? Also localhost requires DNS resolution, which sometimes gets broken when you don't have an internet connection, whereas 127.0.0.1 always works as expected.

Reply:
Also if we assume it always returns localhost, what would be the advantage over 127.0.0.1? It _should_ resolve to the same thing, but why go through DNS?

Reply:
Whatever's using it will probably call `getHostAddress`, which will return `"127.0.0.1"`, so no need for a lookup there. It also means it'll use `"::1"` on IPv6 compatible systems. And `InetSocketAddress` will convert your string into an `InetAddress` anyway, so you might as well save it the effort and give it one already.

My main reasoning is because I don't like magic numbers and I prefer when you can use a meaningfully-named constant, but I am not hugely passionate about this, so by all means, feel free to keep it as it is. 🙂